cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -921,138 +921,164 @@ PyMODINIT_FUNC PyInit_simulator(void)
             "get_definition_file($self)\n"
             "--\n\n"
-            "Get the file that sources the object's definition"
+            "get_definition_file(self) -> str\n"
+            "Get the file that sources the object's definition."
         )
     },
     {"get_handle_by_name",
         (PyCFunction)get_handle_by_name, METH_VARARGS, PyDoc_STR(
             "get_handle_by_name($self, name, /)\n"
             "--\n\n"
-            "Get handle of a named object"
+            "get_handle_by_name(self, name: str) -> cocotb.simulator.gpi_sim_hdl\n"
+            "Get a handle to a child object by name."
         )
     },
     {"get_handle_by_index",
         (PyCFunction)get_handle_by_index, METH_VARARGS, PyDoc_STR(
             "get_handle_by_index($self, index, /)\n"
             "--\n\n"
-            "Get handle of a object at an index in a parent"
+            "get_handle_by_index(self, index: int) -> cocotb.simulator.gpi_sim_hdl\n"
+            "Get a handle to a child object by index."
         )
     },
     {"get_name_string",
         (PyCFunction)get_name_string, METH_NOARGS, PyDoc_STR(
             "get_name_string($self)\n"
             "--\n\n"
-            "Get the name of an object as a string"
+            "get_name_string(self) -> str\n"
+            "Get the name of an object as a string."
         )
     },
     {"get_type_string",
         (PyCFunction)get_type_string, METH_NOARGS, PyDoc_STR(
             "get_type_string($self)\n"
             "--\n\n"
-            "Get the type of an object as a string"
+            "get_type_string(self) -> str\n"
+            "Get the GPI type of an object as a string."
         )
     },
     {"get_type",
         (PyCFunction)get_type, METH_NOARGS, PyDoc_STR(
             "get_type($self)\n"
             "--\n\n"
-            "Get the type of an object, mapped to a GPI enumeration"
+            "get_type(self) -> int\n"
+            "Get the GPI type of an object as an enum."
         )
     },
     {"get_const",
         (PyCFunction)get_const, METH_NOARGS, PyDoc_STR(
             "get_const($self)\n"
             "--\n\n"
-            "Get a flag indicating whether the object is a constant"
+            "get_const(self) -> bool\n"
+            "Return ``True`` if the object is a constant."
         )
     },
     {"get_num_elems",
         (PyCFunction)get_num_elems, METH_NOARGS, PyDoc_STR(
             "get_num_elems($self)\n"
             "--\n\n"
-            "Get the number of elements contained in the handle"
+            "get_num_elems(self) -> int\n"
+            "Get the number of elements contained in the handle."
         )
     },
     {"get_range",
         (PyCFunction)get_range, METH_NOARGS, PyDoc_STR(
             "get_range($self)\n"
             "--\n\n"
-            "Get the range of elements (tuple) contained in the handle, returns None if not indexable"
+            "get_range(self) -> Tuple[int, int]\n"
```

Comment:
```suggestion
            "get_range(self) -> typing.Tuple[int, int]\n"
```
Possibly

Reply:
Unfortunately it's not linking. I think `Tuple` is well understood.